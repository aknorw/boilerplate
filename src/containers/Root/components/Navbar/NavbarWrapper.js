import styled from 'styled-components'

export default styled.nav`
  flex-shrink: 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2rem;
  background-color: black;
  color: white;
`
