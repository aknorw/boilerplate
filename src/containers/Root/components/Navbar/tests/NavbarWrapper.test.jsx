import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import NavbarWrapper from '../NavbarWrapper'

describe('<NavbarWrapper />', () => {
  it('should be defined', () => {
    expect(NavbarWrapper).toBeDefined()
  })
  it('should render correctly', () => {
    const tree = shallow(
      <NavbarWrapper />,
    )
    expect(tree).toMatchSnapshot()
  })
  it('should have style rules', () => {
    const tree = shallow(
      <NavbarWrapper />,
    )
    expect(tree).toHaveStyleRule('flex-shrink', '0')
    expect(tree).toHaveStyleRule('display', 'flex')
    expect(tree).toHaveStyleRule('justify-content', 'space-between')
    expect(tree).toHaveStyleRule('align-items', 'center')
    expect(tree).toHaveStyleRule('padding', '1rem 2rem')
    expect(tree).toHaveStyleRule('background-color', 'black')
    expect(tree).toHaveStyleRule('color', 'white')
  })
})
