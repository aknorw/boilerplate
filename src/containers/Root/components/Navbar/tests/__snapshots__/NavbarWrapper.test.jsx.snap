// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarWrapper /> should render correctly 1`] = `
.c0 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 2rem;
  background-color: black;
  color: white;
}

<nav
  className="c0"
/>
`;
